src/app/backend/handler/localehandler.go

```diff
@@ -110,6 +110,16 @@ func (handler *LocaleHandler) determineLocalizedDir(locale string) string {
 	return defaultDir
 }
 
+func isMatchedLocale(locale string, tag language.Tag) bool {
+	base, _ := tag.Base()
+	if locale != strings.ToLower(tag.String()) {
+		if locale != base.String() {
```

Comment:
To check base I'd extract actual base from locale as we don't know if locale is `zh` or `zh-tw`. You can use `strings.Split(locale, "-")[0]`. For both `zh` and `zh-tw` it would return `zh` as base.

Reply:
@floreks The following code is use `language. NewMatcher` to handle locale:
https://github.com/kairen/dashboard/commit/8787a35448df6c17913ab3f3fcbbba5020ea6b94

Reply:
But this way need to modify `localehandler_test.go` too.

Reply:
I'm not fully sure how NewMatcher actually works. We need behavior such as:

Given supported languages `en, zh-tw, zh-hk, zh`:
 - when required language is `en` return `en`
 - when required language is `zh` return `zh`
 - when required language is `zh-hk` return `zh-hk`
 - when required language is `zh-cn` return `zh`

1. For any required language that is a regional dialect try to find exact match. In case not found: 
   1. return base language if supported
   1. otherwise return default language

Reply:
@floreks I implemented a simple code to check the locale. In the code, You can see that the matcher returns a match tag and confidence value. If the confidence value is `Exact`, It is the best locale. If the confidence value is not `Exact`, we could check the base language, otherwise return default language.

The code as below:
https://gist.github.com/kairen/5ed330dd203f23bca8a29b224a55cf1c

Reply:
I have checked it. Sounds good to me.

Reply:
@floreks The following is a test branch, I have completed the above function changes.
https://github.com/kairen/dashboard/commit/a67118b1e78f239bc3c1b87db511c5f06fc17c29

The test results are as follows:
[Backend test result](https://gist.github.com/kairen/98b578524c2e47dddccbae82c04f4291)

Reply:
Looks good! 

PS. You can work on branch you have used to create PR. There is no need to use other branch.

Reply:
@floreks I have updated to the current branch, Could you check the branch changes?